p2panda-auth/src/group/resolver.rs

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use std::collections::HashSet;
+use std::fmt::Display;
+use std::{fmt::Debug, marker::PhantomData};
+
+use thiserror::Error;
+
+use crate::group::{GroupControlMessage, GroupState};
+use crate::traits::{GroupStore, IdentityHandle, Operation, OperationId, Ordering, Resolver};
+
+use super::GroupStateInner;
+
+// TODO: introduce all error types.
+#[derive(Debug, Error)]
+pub enum GroupResolverError {}
```

Comment:
Nitpick: I found code easier to read which starts with the most concrete implementation at the top, error types seem less important